### PR TITLE
Introduce schema for package tagging tasks

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

**Comment by fridex** [comment]
[test]

File: f8a_worker/workers/schemas/keywords_tagging.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class KeywordsTaggingDetails(jsl.Document):
+    class Options(object):
+        definition_id = "keywords_tagging_details"
+        description = "Details computed by KeywordsTagging worker"
+
+    description = jsl.DictField(required=False, additional_properties=True)
+    # Keywords extracted in mercator (e.g. keywords stated in setup.py)
+    keywords = jsl.ArrayField(required=False, additional_properties=True)
```
Comment (jpopelka) [inline f8a_worker/workers/schemas/keywords_tagging.py]: I'm seeing

```python
   File "/usr/lib/python3.4/site-packages/jsonschema/validators.py", line 130, in validate
     raise error
 jsonschema.exceptions.ValidationError: None is not of type 'array'

 Failed validating 'type' in schema['properties']['details']['properties']['keywords']:
     OrderedDict([('type', 'array')])

 On instance['details']['keywords']:
     None
```


If `keywords` can be `None`, then this should look like:
`keywords = jsl.OneOfField([jsl.ArrayField(required=False, additional_properties=True)], jsl.NullField())`

Comment (fridex) [reply f8a_worker/workers/schemas/keywords_tagging.py]: Thanks! Could you please provide me ecosystem/package/version triplet for which this occurred? This is picked directly from mercator task, so having an empty array makes more sense to me from consistency POV.

Comment (jpopelka) [reply f8a_worker/workers/schemas/keywords_tagging.py]: sorry, `pypi:six:1.10.0`

Comment (fridex) [reply f8a_worker/workers/schemas/keywords_tagging.py]: Thanks, should be fixed. Nice catch 😉

Comment (jpopelka) [reply f8a_worker/workers/schemas/keywords_tagging.py]: `jsl.ArrayField` doesn't seem to have a `additional_properties` argument. As I stated earlier, if `keywords` is a list of strings, then this should be defined as: `keywords = jsl.ArrayField(jsl.StringField())`

File: f8a_worker/workers/schemas/package_keywords_tagging.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+class PackageKeywordsTaggingDetails(jsl.Document):
+    class Options(object):
+        definition_id = "package_keywords_tagging_details"
+        description = "Details computed by PackageKeywordsTagging worker"
+
+    README = jsl.DictField(required=False, dditional_properties=True)
+    package_name = jsl.DictField(required=False, additional_properties=True)
+    repository_description = jsl.DictField(required=False, additional_properties=True)
+    gh_topics = jsl.ArrayField(required=False, additional_properties=True)
```
Comment (jpopelka) [inline f8a_worker/workers/schemas/package_keywords_tagging.py]: what about
`gh_topics = jsl.ArrayField(jsl.StringField(), required=False)`

File: f8a_worker/workers/schemas/keywords_tagging.py (hunk 1 above)
Comment (fridex) [reply f8a_worker/workers/schemas/keywords_tagging.py]: Thanks, hopefully now :)

Comment (jpopelka) [reply f8a_worker/workers/schemas/keywords_tagging.py]: Still seeing the same problem with `keywords` being `None`.
I think we need to make sure that [DataNormalizer._split_keywords()](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/data_normalizer.py#L245) returns always a list, i.e. change
```python
if keywords is None:
    return None
```
to
```python
if keywords is None:
    return []
```

**Comment by jpopelka** [comment]
Tests need to be amended too.

**Comment by jpopelka** [comment]
I have a "ping-pong-ish" feeling from this review Frido ;-)

**Comment by fridex** [comment]
> I have a "ping-pong-ish" feeling from this review Frido ;-)

I will need to move to my silence hanging egg and check this in a piece environment without disturbance... :)

**Comment by fridex** [comment]
@jpopelka green, PTAL :)

**Pull request merged**